src/qibolab/pulses.py

```diff
@@ -1400,7 +1403,15 @@ def modulated_waveforms(self):  #  -> tuple[Waveform, Waveform]:
 
     @property
     def serial(self):
-        return f"FluxPulse({self.start}, {self.duration}, {format(self.amplitude, '.6f').rstrip('0').rstrip('.')}, {self.shape}, {self.channel}, {self.qubit})"
+        return f"{self.__class__.__name__}({self.start}, {self.duration}, {format(self.amplitude, '.6f').rstrip('0').rstrip('.')}, {self.shape}, {self.channel}, {self.qubit})"
```

Comment:
Ok, this was existing even before this PR (so we will split and discuss in a separate issue), but: can't we convert this to a `dataclass` as well and avoid serializing with a string? (that has to either be parsed or `exec`uted as is)

Reply:
All `Pulse` objects can be converted to `dataclass` but this I would certainly do in a different PR as it may be quite complex, since pulses are used almost everywhere around the code.

Reply:
I believe there is no doubt it won't be done here.